docs/api-reference/deck.md

```diff
@@ -113,6 +113,17 @@ A geospatial `viewState` would typically contain the following fields:
 * `pitch` (Number, optional) - Current pitch - used to define a mercator projection if `viewport` is not supplied.
 
 
+#### `initialViewState` (Object)
+
+If `initialViewState` is provided, the `Deck` component will track view state changes from any attached `controller` using internal state, with `initialViewState` as its initial view state.
+
+Notes:
+* The `props.onViewStateChange` callback will still be called, if provided.
```

Comment:
This design switches the viewport change callback from an "intercept" model to a "notify" model. It makes applying custom viewport constraints more difficult, such as https://github.com/uber/react-map-gl/issues/442. My preference would be using this callback to intercept a view state change, modify state or even cancel the event if needed.

Reply:
Added support for modified return value to the `onViewStateChange` callback, described below.